mapbook-iOS/View controllers/Map/MapViewController.swift

```diff
@@ -110,6 +88,86 @@ class MapViewController: UIViewController {
         
         present(action, animated: true, completion: nil)
     }
+    
+    //MARK:- Search
+    
+    weak var locatorTask: AGSLocatorTask? {
+        didSet {
+            updateBarButtonItems()
+            updateSearchController()
+        }
+    }
+            
+    @objc func toggleShouldShowSearch(_ sender: UIBarButtonItem) {
+        shouldShowSearch.toggle()
+    }
+    
+    private var shouldShowSearch = false {
+        didSet {
+            updateSearchController()
+        }
+    }
+      
+    private func updateBarButtonItems() {
+        if locatorTask == nil {
+            navigationItem.rightBarButtonItems = [ellipsisButton]
+        }
+        else {
+            let searchButton = UIBarButtonItem(
+                image: UIImage(named: "search"),
+                style: .plain,
+                target: self,
+                action: #selector(toggleShouldShowSearch)
+            )
+            navigationItem.rightBarButtonItems = [ellipsisButton, searchButton]
+        }
+    }
+    
+    private func updateSearchController() {
+        if shouldShowSearch, let locatorTask = locatorTask {
+            let results = LocatorSuggestionController.fromStoryboard(with: locatorTask)
+            results.delegate = self
+            navigationItem.searchController = {
+                let controller = UISearchController(searchResultsController: results)
+                controller.searchResultsUpdater = results
+                controller.searchBar.placeholder = "Search the map."
+                return controller
+            }()
+        }
+        else {
+            navigationItem.searchController = nil
+        }
+        //refresh navigation controller layout.
+        navigationController?.view.setNeedsLayout()
+    }
+    
+    private func showSearch(result: AGSGeocodeResult) {
+        shouldShowSearch.toggle()
+        if let location = result.displayLocation, let extent = result.extent {
+            if #available(iOS 13.0, *) {
+                mapView.callout.accessoryButtonType = .close
```

Comment:
Ah, I see what you're saying here.

Unfortunately we have a problem... and it's a bit counter-intuitive. I think your suggestion is rooted in the notion that the Locator and the Feature Layers are tightly coupled. As I understand it, it is a mere (in)convenience that the data overlaps. This is because this data set was used to generate a Locator from content found in the Feature Layers. What this means is the Search Controller is performing a search on the Locator and not the Feature Layers. Because these two data sources are disconnected, searching the Locator does not produce a pop-up (or a Feature for that matter).

Now, I could have the callout `information` accessory button perform an identify for features at the callout's geometry. The problem with this technique is that we shouldn't assume features in other data sets are also loosely and (in)conveniently coupled to one another.

Do you see the problem here? I'm open to finding a way to improve the story, but i'm not sure this is the way to do it... What do you think?

Reply:
Ok, I get it.  It's fine as it is then.